src/test/java/org/wso2/carbon/connector/integrationTest/SMPP/SMSConfigTest.java

```diff
@@ -0,0 +1,171 @@
+import static org.powermock.api.mockito.PowerMockito.*;
+
+import org.powermock.api.mockito.PowerMockito;
+import org.wso2.carbon.esb.connector.SendSMS;
+
+import java.util.Stack;
+
+@PrepareForTest({org.wso2.carbon.esb.connector.SMSConfig.class, SendSMS.class})
+
+public class SMSConfigTest {
+
+    private MessageContext ctx;
+    ConfigurationContext configContext;
+    SynapseConfiguration synapseConfig;
+    SMSConfig smsConfig;
+    SendSMS sendSMS;
+    private TemplateContext templateContext;
+    private Stack functionStack;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        ctx = createMessageContext();
+        smsConfig = new SMSConfig();
+        sendSMS = new SendSMS();
+        templateContext = new TemplateContext("smpp", null);
+        functionStack = new Stack();
+        initMocks(this);
+    }
+
+    @Test
+    public void testConnect() throws Exception {
+        templateContext.getMappedValues().put(SMPPConstants.HOST, "127.0.0.1");
+        templateContext.getMappedValues().put(SMPPConstants.SYSTEM_ID, "test");
+        templateContext.getMappedValues().put(SMPPConstants.PASSWORD, "pass");
+        templateContext.getMappedValues().put(SMPPConstants.PORT, "2775");
+        templateContext.getMappedValues().put(SMPPConstants.ENQUIRELINK_TIMER, "1000");
+        templateContext.getMappedValues().put(SMPPConstants.TRANSACTION_TIMER, "80");
+        templateContext.getMappedValues().put(SMPPConstants.SYSTEM_TYPE, "UNKNOWN");
+        templateContext.getMappedValues().put(SMPPConstants.ADDRESS_TON, "NETWORK_SPECIFIC");
+        templateContext.getMappedValues().put(SMPPConstants.ADDRESS_NPI, "INTERNET");
+        functionStack.push(templateContext);
+        ctx.setProperty("_SYNAPSE_FUNCTION_STACK", functionStack);
+
+        SMPPSession session = PowerMockito.mock(SMPPSession.class);
+        BindParameter bindParameter = PowerMockito.mock(BindParameter.class);
+        whenNew(SMPPSession.class).withNoArguments().thenReturn(session);
+        whenNew(BindParameter.class).withAnyArguments().thenReturn(bindParameter);
+        when(session.connectAndBind(anyString(), anyInt(), any(BindParameter.class))).thenReturn("some string");
+
+        smsConfig.connect(ctx);
+        Assert.assertEquals(session, ctx.getProperty(SMPPConstants.SMPP_SESSION));
+    }
+
+    @Test(expectedExceptions = SynapseException.class)
+    public void testConnectNegativeMissingHostValue() throws Exception {
+        templateContext.getMappedValues().put(SMPPConstants.PORT, "2775");
+        templateContext.getMappedValues().put(SMPPConstants.SYSTEM_ID, "test");
+        templateContext.getMappedValues().put(SMPPConstants.PASSWORD, "pass");
+        functionStack.push(templateContext);
+        ctx.setProperty("_SYNAPSE_FUNCTION_STACK", functionStack);
+        smsConfig.connect(ctx);
+    }
+
+    @Test(expectedExceptions = SynapseException.class)
+    public void testConnectNegativeMissingSystemIdValue() throws Exception {
```

Comment:
no assets to test this?

Reply:
We can't do asset. Because throwing an exception. So I checked like this
@Test(expectedExceptions = SynapseException.class)

Reply:
ok